### PR TITLE
Extract signal handling from service.Service

### PR DESCRIPTION
**Pull request by russellhaering**
Vulcand is well factored for use as a library. The main pain point I've hit is that once you've called `Start()` on a service there is no way to stop it - it directly registers a bunch of signal handlers which are used to trigger shutdown.

This change abstracts signals into `ControlCode`s, a channel of which must be passed to `Start()`. The behavior of `service.Run()` is unchanged - signal handlers are registered as before, and each signal is mapped to the corresponding control code.

### CODE REVIEW

**Comment by klizhentas** [comment]
Good idea @russellhaering feel free to merge this in

File: service/service.go [hunk 1]
```diff
@@ -31,13 +31,47 @@ import (
...
+		for {
+			signal := <-sigC
+			log.Infof("Got signal '%s'", signal)
+
+			switch signal {
+			case syscall.SIGTERM, syscall.SIGINT:
+				controlC <- ControlCodeGracefulShutdown
+			case syscall.SIGKILL:
+				controlC <- ControlCodeImmediateShutdown
+			case syscall.SIGUSR2:
```
Comment (gchaincl) [inline service/service.go]: why don't you handle this along with `case syscall.SIGTERM, syscall.SIGINT`

Comment (russellhaering) [inline service/service.go]: Good catch, this case had a typo: it was sending `ControlCodeGracefulShutdown` wghen it should have been sending `ControlCodeForkChild`. Fixed in aa74c54.

Comment (gchaincl) [inline service/service.go]: :+1:

**Pull request merged**